Emby.Server.Implementations/HttpClientManager/HttpClientManager.cs

```diff
@@ -74,146 +74,122 @@ public class HttpClientManager : IHttpClient
         /// DON'T dispose it after use.
         /// </summary>
         /// <value>The HTTP clients.</value>
-        private readonly ConcurrentDictionary<string, HttpClientInfo> _httpClients = new ConcurrentDictionary<string, HttpClientInfo>();
+        private readonly ConcurrentDictionary<string, HttpClient> _httpClients = new ConcurrentDictionary<string, HttpClient>();
 
         /// <summary>
         /// Gets
         /// </summary>
-        /// <param name="host">The host.</param>
+        /// <param name="url">The host.</param>
         /// <param name="enableHttpCompression">if set to <c>true</c> [enable HTTP compression].</param>
         /// <returns>HttpClient.</returns>
         /// <exception cref="ArgumentNullException">host</exception>
-        private HttpClientInfo GetHttpClient(string host, bool enableHttpCompression)
+        private HttpClient GetHttpClient(string url, bool enableHttpCompression)
         {
-            if (string.IsNullOrEmpty(host))
-            {
-                throw new ArgumentNullException(nameof(host));
-            }
-
-            var key = host + enableHttpCompression;
+            var key = GetHostFromUrl(url) + enableHttpCompression;
 
             if (!_httpClients.TryGetValue(key, out var client))
```

Comment:
Shouldn't we be using the HttpClientFactory?

Reply:
Maybe in the future? 🤷‍♂️